runtime_magic.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2022, The Garble Authors.
+// See LICENSE for licensing information.
+
+package main
+
+import (
+	"go/ast"
+	"go/token"
+	"strconv"
+)
+
+// updateMagicValue updates hardcoded value of hdr.magic
+// when verifying header in symtab.go
+func updateMagicValue(file *ast.File, magicValue uint32) {
+	// Find `hdr.magic != 0xfffffff?` in symtab.go and update to random magicValue
+	updateMagic := func(node ast.Node) bool {
+		binExpr, ok := node.(*ast.BinaryExpr)
+		if !ok || binExpr.Op != token.NEQ {
+			return true
+		}
+
+		selectorExpr, ok := binExpr.X.(*ast.SelectorExpr)
+		if !ok {
+			return true
+		}
+
+		if ident, ok := selectorExpr.X.(*ast.Ident); !ok || ident.Name != "hdr" {
+			return true
+		}
+		if selectorExpr.Sel.Name != "magic" {
+			return true
+		}
+
+		if _, ok := binExpr.Y.(*ast.BasicLit); !ok {
+			return true
+		}
+		binExpr.Y = &ast.BasicLit{
+			Kind:  token.INT,
+			Value: strconv.FormatUint(uint64(magicValue), 10),
+		}
+		return false
+	}
+
+	ast.Inspect(file, updateMagic)
```

Comment:
nit: you're walking the entire file, which is ~37KiB of Go code. each walk is relatively expensive, as you are walking the AST for the entire file.

it would be significantly cheaper if you ranged over the file's declarations, looked for a `FuncDecl` whose `Name` is `moduledataverify1`, and then walked its body only.

Reply:
Also, this function should error or panic if it fails to find the function and modify the magic value. Otherwise, if upstream Go changes in some way that breaks our logic (e.g. refactoring the equality check), `garble build` would happily continue without altering the magic value, producing a bad binary.